### PR TITLE
Adds golintCI to travis job

### LINKED ISSUE #1927: golangci-lint reports dead code, missing error checks and other issues.
golangci-lint is a meta-linter that runs a bunch of other linting tools (lint, vet etc.) very quickly.
With default settings it catches a lot of real problems and very few false ones.
I recommend we included it in the odo build vaildation.

## Actual behavior

In root of odo repo:
```
go get github.com/golangci/golangci-lint/cmd/golangci-lint
golangci-lint run ./...
```
Prints a lot of  issues

## Expected behavior

No issues.

## Any logs, error output, etc?

[lint.txt](https://github.com/openshift/odo/files/3412381/lint.txt)

### PR DESCRIPTION
**Pull request by mik-dass**
**What kind of PR is this?**

/kind cleanup

**What does does this PR do / why we need it:**

It adds golangci-lint as a travis job
It also adds the golangci-lint as make target.
It also fixes the current lint errors in our code base.

**Which issue(s) this PR fixes:**

Fixes #1927

**How to test changes / Special notes to the reviewer:**

run `golangci-lint run` and check that no errors occur

### CODE REVIEW

File: pkg/component/watch_test.go [hunk 1]
```diff
@@ -80,22 +79,17 @@ var StartChan = make(chan bool)
...
 				ExtChan <- true
 				return nil
 			}
 		}
-		if !found {
-			ExtChan <- true
-			return fmt.Errorf("received %+v which is not same as expected list %+v", files, ExpectedChangedFiles)
-		}
 	}
+	ExtChan <- true
```
Comment (mik-dass) [inline pkg/component/watch_test.go]: @kadel Please have a look at this part.

File: pkg/component/watch_test.go [hunk 2]
```diff
@@ -233,42 +227,37 @@ func TestWatchAndPush(t *testing.T) {
...
 				t.Logf("Starting file simulations \n%+v\n", tt.fileModifications)
 				// Simulating file modifications for watch to observe
-				for {
-					select {
-					case startMsg := <-StartChan:
-						if startMsg {
-							for _, fileModification := range tt.fileModifications {
+				startMsg := <-StartChan
+				if startMsg {
+					for _, fileModification := range tt.fileModifications {
```
Comment (mik-dass) [inline pkg/component/watch_test.go]: @kadel Please have a look at this too

**Comment by cdrage** [comment]
Git commit message is blank and the PR description is very limited. Please update them both!

/hold

Screenshot from 2019-11-01 11-19-07

**Comment by mik-dass** [comment]
> Git commit message is blank and the PR description is very limited. Please update them both!

@cdrage Fixed

**Comment by mik-dass** [comment]
```
E1104 09:01:02.793480   16615 occlient.go:2664] Command 'tar xf - -C /tmp/src --strip 1' in container failed.
[odo] E1104 09:01:02.793514   16615 occlient.go:2665] stdout: 
[odo] E1104 09:01:02.793520   16615 occlient.go:2666] stderr: tar: /tmp/src: Cannot open: No such file or directory
[odo] tar: Error is not recoverable: exiting now
[odo] 
```
/retest

**Comment by mik-dass** [comment]
```
Running oc with args [oc get pods --namespace drcyasvliu --selector=deploymentconfig=nodejs-push-atonce-nodejs-create-now-test -o jsonpath='{.items[0].metadata.name}']
[oc] 'nodejs-push-atonce-nodejs-create-now-test-1-6blpv'Running oc with args [oc exec nodejs-push-atonce-nodejs-create-now-test-1-6blpv --namespace drcyasvliu -c nodejs-push-atonce-nodejs-create-now-test -- ls -la /tmp/src/package.json]
[oc] ls: cannot access /tmp/src/package.json: No such file or directory
```
/retest

**Comment by mik-dass** [comment]
```
Caching blobs under "/var/cache/blobs".
[odo] Warning: Pull failed, retrying in 5s ...
[odo] Warning: Pull failed, retrying in 5s ...
[odo] Warning: Pull failed, retrying in 5s ...
[odo] error: build error: After retrying 2 times, Pull image still failed due to error: while pulling "docker://image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb" as "image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb": Error initializing source docker://image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb: pinging docker registry returned: Get https://image-registry.openshift-image-registry.svc:5000/v2/: dial tcp 172.30.13.161:5000: connect: no route to host
```
/retest

**Comment by amitkrout** [comment]
> ```
> Caching blobs under "/var/cache/blobs".
> [odo] Warning: Pull failed, retrying in 5s ...
> [odo] Warning: Pull failed, retrying in 5s ...
> [odo] Warning: Pull failed, retrying in 5s ...
> [odo] error: build error: After retrying 2 times, Pull image still failed due to error: while pulling "docker://image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb" as "image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb": Error initializing source docker://image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:d30db914ff173925adb9e3a65d61e863e19ea679d80fe91ab6ac3c99bc5c5ccb: pinging docker registry returned: Get https://image-registry.openshift-image-registry.svc:5000/v2/: dial tcp 172.30.13.161:5000: connect: no route to host
> ```
> 
> /retest

@mik-dass This is network glitch. I have seen similar error in my previous minishift project

**Comment by mik-dass** [comment]
```
 Timed out after 600.000s.
    Running odo with args [odo service create -w dh-postgresql-apb --project vbnoqraanu --plan dev -p postgresql_user=luke -p postgresql_password=secret -p postgresql_database=my_data -p postgresql_version=9.6]
    Expected process to exit.  It did not.
```
/retest

File: pkg/component/watch_test.go (hunk 1 above)
Comment (kadel) [reply pkg/component/watch_test.go]: I'm not sure about this at all :-(

Comment (mik-dass) [reply pkg/component/watch_test.go]: After debugging I found that `found = true` was never called. `found` always stayed false and thus I removed the variable completely.
```
ineffectual assignment to `found` (ineffassign)
```
This is the error reported by golangci-lint. Also if if `found=true` does get executed, we are returning from the function after that, so changing the found value is useless.

File: pkg/component/watch_test.go [hunk 3]
```diff
@@ -80,22 +79,17 @@ var StartChan = make(chan bool)
...
 				return nil
 			}
 		}
-		if !found {
-			ExtChan <- true
-			return fmt.Errorf("received %+v which is not same as expected list %+v", files, ExpectedChangedFiles)
-		}
 	}
+	ExtChan <- true
 	return nil
```
Comment (kadel) [inline pkg/component/watch_test.go]: The error should be returned here

File: pkg/component/watch_test.go (hunk 1 above)
Comment (kadel) [reply pkg/component/watch_test.go]: that makes sense

File: pkg/component/watch_test.go (hunk 2 above)
Comment (kadel) [reply pkg/component/watch_test.go]: I think that the `for` loop is needed, as it is being used to wait for the data from the `StartChan`

**Pull request merged**